web/client/utils/grids/GridLabelsUtils.js

```diff
@@ -0,0 +1,128 @@
+import isFunction from "lodash/isFunction";
+
+/**
+ * Returns the modulo of a / b, depending on the sign of b.
+ *
+ * @param {number} a Dividend.
+ * @param {number} b Divisor.
+ * @return {number} Modulo.
+ */
+function modulo(a, b) {
+    var r = a % b;
+    return r * b < 0 ? r + b : r;
```

Comment:
((a % b ) + b ) % b should do the same, I think :smile:  No problem to do it more explicit. 
ref: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Remainder